src/averageAges.js

```diff
@@ -55,7 +80,44 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let totalAgeDiff = 0;
+  let count = 0;
+
+  if (onlyWithSon) {
+    people.forEach(person => {
+      if (person.sex === 'm' && (!onlyWithSon || hasMother(person, people))) {
+        const mother = findMother(person, people);
+
+        if (mother) {
+          totalAgeDiff += person.born - mother.born;
+          count++;
+        }
+      }
+    });
+  } else {
+    people.forEach(person => {
+      if (!onlyWithSon || hasMother(person, people)) {
+        const mother = findMother(person, people);
+
+        if (mother) {
+          totalAgeDiff += person.born - mother.born;
+          count++;
+        }
+      }
+    });
+  }
+
+  const tot = totalAgeDiff / count;
+
+  return count === 0 ? 0 : tot;
```

Comment:
Instead of checking if count equals 0 at the end, check at the start if the array is empty and if it is, return 0.